config/pmd-test.xml

```diff
@@ -140,7 +140,9 @@
                    | //ClassOrInterfaceDeclaration[@SimpleName='ArchUnitSuperClassTest']
                        //MethodDeclaration
                    | //ClassOrInterfaceDeclaration[@SimpleName='ArchUnitCyclesCheckTest']
-                       //MethodDeclaration"/>
+                       //MethodDeclaration
+                   | //ClassOrInterfaceDeclaration[@SimpleName='SuppressionFilterTest']
+                       //MethodDeclaration//Name[starts-with(@SimpleName, 'testUseCache')]"/>
```

Comment:
@shubh220922 try to dump AST of the file to help you build/validate your XPath query: https://pmd.github.io/latest/pmd_devdocs_experimental_ast_dump.html

Reply:
PMD has even UI tool to test xpath, please play with it.

Reply:
try
`//MethodDeclaration[starts-with(@SimpleName, 'testUseCache')]"/>`

Reply:
@romani 
build still failed after writing ```//MethodDeclaration[starts-with(@SimpleName, 'testUseCache')]"/>```
![image](https://user-images.githubusercontent.com/90137881/210445935-49902489-a169-4277-a51d-c7f52ec7bcf9.png)
Still showing the same error as before. 
```
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest:163 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest:179 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
```

Reply:
@nrmancuso 

- [x] Downloaded the necessary softwares.
- [x] Dumped the AST of ```SuppressionFilterTest``` in an xml file.
- [ ] Attempted validating query multiple times but command seems to fail (CLI below):

```
PS C:\Users\shubh> xml --version
1.6.1
compiled against libxml2 2.9.1, linked with 20901
compiled against libxslt 1.1.28, linked with 10128
PS C:\Users\shubh> xml select -t -c "//ClassOrInterfaceDeclaration[@SimpleName='SuppressionFilterTest']//MethodDeclaration[starts-with(@SimpleName, 'testUseCache')]" "C:\Users\shubh\OneDrive\Desktop\checkstyle project\foo.xml"
PS C:\Users\shubh> xml select -t -c "//MethodDeclaration[starts-with(@SimpleName, 'testUseCache')]" "C:\Users\shubh\OneDrive\Desktop\checkstyle project\foo.xml"                                                                                                    
PS C:\Users\shubh> xml select -t -c "//ClassOrInterfaceDeclaration[@SimpleName='SuppressionFilterTest']//MethodDeclaration//Name[starts-with(@SimpleName, 'testUseCache')]" "C:\Users\shubh\OneDrive\Desktop\checkstyle project\foo.xml"
PS C:\Users\shubh> xml select -t -c "//MethodDeclaration//Name[starts-with(@SimpleName, 'testUseCache')]" "C:\Users\shubh\OneDrive\Desktop\checkstyle project\foo.xml"
```

Here expected was to get an output after execution of the commands above but was found no output. Am I querying the wrong way?

I am using xmlstarlet for query.

Reply:
Please share the generated xml file by GitHub gist.

Reply:
@nrmancuso here it is:

https://gist.githubusercontent.com/shubh220922/3db3a54bbc2edb9e77166274c4ee7ed1/raw/e757edaa045f4af2ada754f965feb2ae98915066/dump.xml

Reply:
```
<MethodDeclaration Abstract='false' Arity='0' Default='false' Final='false' Image='' InterfaceMember='false' Kind='METHOD' MethodName='testUseCacheLocalFileExternalResourceContentDoesNotChange' Modifiers='1' Name='testUseCacheLocalFileExternalResourceContentDoesNotChange' Native='false' PackagePrivate='false' Private='false' Protected='false' Public='true' Static='false' Strictfp='false' Synchronized='false' SyntacticallyAbstract='false' SyntacticallyPublic='true' Transient='false' Void='true' Volatile='false'>
```
I do not see `SimpleName` attribute in this node, only `Name` and `MethodName`.

Reply:
@nrmancuso below:

https://gist.githubusercontent.com/shubh220922/72138e2b956a0b7a49799c0c47df9379/raw/34ff3ad31a37ec28ac1887d875a23ea79731c1a5/query.xml

Reply:
> I do not see `SimpleName` attribute in this node, only `Name` and `MethodName`.

Thanks for pointing out. 

now it is working.. i have shared the gist of the output @nrmancuso

Reply:
Let's see if suppression works now  :)

Reply:
@shubh220922 suppression seems successful. Maven = Success. Checking CI.